src/generate/caro/evalFunctions_caro.hpp

```diff
@@ -72,7 +112,9 @@ class evalFunctions_caro {
         static double numObservation;
 
         static double maxChipsInCasino;
+        static double maxIpInCasino;
         static double chipsToIpJudge;
+        static double secretsToIpJudge;
```

Comment:
ich habe den eindruck, dass das klassenvariablen sein sollten und es wenig sinn macht diese klasse völlig static zu machen.

Reply:
Du meinst, dass ich die ganzen static variablen als private class members machen soll und die dann im Konstruktor initialisieren?

Reply:
Aber wo initialisiere ich die Klasse dann?
Weil die static Variablen bleiben eigentlich während des gesamten Spiels (also Item, Equipment und GameOperation) gleich.
Ich könnte das zwar in AI.cpp erstellen und als Paremeter übergen, aber für die random "Strategie" brauche ich das ja gar nicht.
Und für jeden Request das Objekt neu zu erstellen ist auch nicht sinnvoll.

Reply:
Also initialisierung kann in-class oder im ctor gemacht werden. Werden die momentan überhaupt irgendwo initialisiert? Ist mir nicht ersichtlich.

Instanziiert hätte ich das in der AI (als membervariable), dann wird das nur 1 mal beim start der AI initialisiert. (RAII pattern, das initialisieren ist keine extra funktion)

Reply:
Ok, ich denke dass ich dafür eine Art wrapper Klasse mache, die ein variant von verschiedenen "helper typen" hat und je nachdem welche Difficulty gewählt ist, ist in dem variant eine Klasse eines anderen Typs. Dann hat man immer die gleiche Parameterliste, egal welcher Ansatz.
wird nicht in diesem pr gemacht, see #48